### PR TITLE
Feature wordsearch

### PR DESCRIPTION
**Pull request by EXR-97**

### CODE REVIEW

File: src/PerformWordSearchOperation.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+            fileContent = Files.readString(Path.of(filepath));
+            if (fileContent.isEmpty()) {
+                System.out.println("File is Empty");
+                return;
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        fileContent = fileContent.replaceAll("[^a-zA-Z0-9@-]", " ");
+        System.out.println("Enter the word to search");
```
Comment (premsrj) [inline src/PerformWordSearchOperation.java]: Requirement is that the filename and word to search will be passed in as command line arguments.

File: src/WordSearch.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        WordSearch wordsearch = new WordSearch();
+        wordsearch.filePath(filepath);
+    }
+
+    /*
+    Assigning the file path to File and checking file is Present in the system or not
+    */
+    public void filePath(String filepath) throws Exception {
+        System.out.println("Searching File...");
+        Thread.sleep(500);
```
Comment (premsrj) [inline src/WordSearch.java]: What is the need for this Thread.sleep() call?

File: src/WordSearch.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    public static void main(String[] args) throws Exception {
+        String filepath = args[0];
+        WordSearch wordsearch = new WordSearch();
+        wordsearch.filePath(filepath);
+    }
+
+    /*
+    Assigning the file path to File and checking file is Present in the system or not
+    */
+    public void filePath(String filepath) throws Exception {
```
Comment (premsrj) [inline src/WordSearch.java]: Give better name for method. searchFile, processFile or something similar

File: src/WordSearch.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import java.io.File;
+
+/**
+ * Getting the file path from the command prompt
+ * calling the filePath method for searching the file in system
+ */
+public class WordSearch {
+    public static void main(String[] args) throws Exception {
+        String filepath = args[0];
```
Comment (premsrj) [inline src/WordSearch.java]: What if no parameter is passed in by the user?

**Review by premsrj — CHANGES_REQUESTED** [review]

File: src/PerformWordSearchOperation.java (hunk 1 above)
Comment (EXR-97) [reply src/PerformWordSearchOperation.java]: Made the changes as per your suggestion.

File: src/WordSearch.java (hunk 3 above)
Comment (EXR-97) [reply src/WordSearch.java]: Renamed the method.

File: src/WordSearch.java (hunk 2 above)
Comment (EXR-97) [reply src/WordSearch.java]: To pause the execution for few milliseconds, But I have removed it now.

File: src/WordSearch.java (hunk 4 above)
Comment (EXR-97) [reply src/WordSearch.java]: Updated and checked the condition as you said.

File: src/WordSearch.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+            wordsearch.processFile(filepath, keyWordToSearch);
+        } else {
+            System.out.println("Please pass the Arguments");
+        }
+    }
+
+    /*
+    Assigning the file path to File and checking file is Present in the system or not
+    */
+    public void processFile(String filepath, String keyWordToSearch) {
```
Comment (mh-scarmel) [inline src/WordSearch.java]: Reform the method like public static void processFile(String filepath, String keyWordToSearch)
once reform not need to call like 
WordSearch wordsearch = new WordSearch();
            wordsearch.processFile(filepath, keyWordToSearch);
just add method not need to initialize an object
processFile(filepath, keyWordToSearch);

File: src/PerformWordSearchOperation.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+        try {
+            fileContent = Files.readString(Path.of(filepath));
+            if (fileContent.isEmpty()) {
+                System.out.println("File is Empty");
+                return;
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        fileContent = fileContent.replaceAll("[^a-zA-Z0-9@-]", " ");
```
Comment (mh-scarmel) [inline src/PerformWordSearchOperation.java]: all logic should be in try catch block

File: src/WordSearch.java [hunk 7]
```diff
@@ -0,0 +1,33 @@
+import java.io.File;
+
+/**
+ * Getting the file path from the command prompt
+ * calling the filePath method for searching the file in system
+ */
+public class WordSearch {
+    public static void main(String[] args) {
+        if (args.length == 2) {
```
Comment (premsrj) [inline src/WordSearch.java]: As per requirement, only txt and json files should be supported. Please add a check for that as well.

**Review by premsrj — CHANGES_REQUESTED** [review]

File: src/WordSearch.java (hunk 5 above)
Comment (EXR-97) [reply src/WordSearch.java]: Updated as Suggested.

File: src/PerformWordSearchOperation.java (hunk 6 above)
Comment (EXR-97) [reply src/PerformWordSearchOperation.java]: Wrote the logic in try block.

File: src/WordSearch.java (hunk 7 above)
Comment (EXR-97) [reply src/WordSearch.java]: Updated to check .txt and .json file format.

**Review by mh-scarmel — APPROVED** [review]

**Review by premsrj — APPROVED** [review]
Will the database implementation be in another PR?

**Review by charanvelury — COMMENTED** [review]
Vinod , The syntax looks of existing code looks fine but when I looked at the requirement email I see that you need insert word search status details into the table called audit whose code I didn't fine. Not sure whether I am missing to see something here.

**Review by charanvelury — APPROVED** [review]

**Pull request merged**